policy/model/helper.go

```diff
@@ -0,0 +1,28 @@
+package model
+
+import (
+	"reflect"
+	"regexp"
+	"strings"
+)
+
+// fieldMap converts a struct into a field name to value map.
+func fieldMap(s interface{}) map[string]interface{} {
+	fields := make(map[string]interface{})
+	e := reflect.ValueOf(s).Elem()
+	typeOfE := e.Type()
+	for i := 0; i < e.NumField(); i++ {
+		f := e.Field(i)
+		name := lowerLeadingUpcases(typeOfE.Field(i).Name)
+		fields[name] = f.Interface()
+	}
+	return fields
+}
+
+var leadingUpcases = regexp.MustCompile("(^[A-Z]*)([^A-Z]*.*)")
+
+func lowerLeadingUpcases(s string) string {
```

Comment:
Could you add some comments about what the expected inputs / outputs are?

Reply:
done